src/mock-generator.ts

```diff
@@ -0,0 +1,181 @@
+    ClassDeclaration,
+    MethodDeclaration,
+    PropertyDeclaration,
+    GetAccessorDeclaration,
+    SetAccessorDeclaration,
+    ParameterDeclaration,
+} from 'ts-simple-ast';
+
+import { ClassInfo } from './types/class-info';
+import { GetAccessorInfo, SetAccessorInfo } from './types/accessor-info';
+import { PropertyInfo, ParameterPropertyInfo } from './types/property-info';
+import { MethodInfo } from './types/method-info';
+import { ClassMember } from './types/class-member';
+import { ClassGenerator } from './class-generator';
+import { REGEX_MARKER_CUSTOM_CODE_BEGIN, REGEX_MARKER_CUSTOM_CODE_END, FILENAME_SUFFIX_MOCK_CLASS } from './constants';
+
+export class MockGenerator {
+
+    private ast: AST;
+    private mockGenerator: ClassGenerator;
+
+    constructor() {
+        this.init();
+    }
+    
+    private init() {
+        this.mockGenerator = new ClassGenerator();
+    }
+    
+    public generate( sourceFilePathPattern: string = './src/**/*.ts' ): void {
+        
+        this.ast = new AST();
+        this.ast.addSourceFiles( sourceFilePathPattern );
+        this.ast.getSourceFiles().forEach( sourceFile => {
+            const sourceClasses = sourceFile.getClasses();
+        
+            if (!sourceClasses.length) {
+                return;
+            }
+        
+            const rootPath = process.cwd().replace( /\\/g, '/' );
+            const sourcePath = sourceFile.getFilePath();
+        
+            const outputPath = sourcePath.replace(rootPath + '/src', rootPath + '/test').replace(/\.ts$/, FILENAME_SUFFIX_MOCK_CLASS);
+            const outputDirname = path.dirname(outputPath);
+            const relativePath = path.relative(outputDirname, sourcePath).replace( /\\/g, '/' );
+        
+            const allUserWrittenCodes = this.getUserWrittenCodeFromMockFile( outputPath );
+        
+            const mockedClasses = sourceClasses
+                .map(sourceClass => {
+                    let userWrittenCodes = allUserWrittenCodes ? allUserWrittenCodes[ 'Mock' + sourceClass.getName() ] : undefined;
+                    return this.createMockClass( sourceClass, userWrittenCodes, outputPath, relativePath ); 
+                })
+                .reduce((acc: string[], next: string[]) => {
+                    return acc.concat(next);
+                }, []);
+        
+            const mockedSource = [
+                '// tslint:disable',
+                '',
+                ...mockedClasses,
+                '',
+            ].join('\n');
+        
+            this.writeToFile( outputDirname, outputPath, mockedSource );
+
+        });
+    }
+
+    private writeToFile( outputDirname: string, outputPath: string, mockedSource: string ) {
+        try {
+            statSync(outputDirname);
+        } catch (err) {
+            mkdirSync(outputDirname);
+        }
+        writeFileSync(outputPath, mockedSource);
+    }
+    
+    
+    private getUserWrittenCodeFromMockFile( filepath: string ) {
```

Comment:
This should go to a generator function. User generated code can be extracted from the sourceClass so no need to read it again. And this should be in the custom code generator branch.